src/accel/cuda/GPUSolver.h

```diff
@@ -14,22 +14,33 @@
 #include "../../Solver.h"
 #endif
 
-#define PySys_WriteStdout printf
-
-#include <thrust/reduce.h>
 #include <thrust/device_vector.h>
+#include <thrust/copy.h>
+#include <thrust/fill.h>
+#include <thrust/reduce.h>
+#include <thrust/replace.h>
+#include <thrust/functional.h>
+#include <thrust/iterator/constant_iterator.h>
 #include <sm_20_atomic_functions.h>
 #include "clone.h"
 #include "GPUExpEvaluator.h"
 
+#define PySys_WriteStdout printf
```

Comment:
You mean put it after the Python.h include? But before the other includes? I get the following error with CUDA 5.5 and 6.5 if I put it before all of the includes:

```bash
nvcc -DFP_PRECISION=double -DDOUBLE -DCUDA -DCCACHE_CC=nvcc -I/usr/local/cuda/include -I/usr/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c src/accel/cuda/GPUSolver.cu -o build/temp.linux-x86_64-2.7/src/accel/cuda/GPUSolver.o --relocatable-device-code true -c -O3 --compiler-options -fpic -gencode=arch=compute_20,code=sm_20 -gencode=arch=compute_30,code=sm_30
/usr/include/python2.7/sysmodule.h(17): error: cannot overload functions distinguished by return type alone

1 error detected in the compilation of "/tmp/tmpxft_0000676b_00000000-9_GPUSolver.compute_30.cpp1.ii".
error: command 'nvcc' failed with exit status 2
```